generators/app/templates/gulp/tasks/templates.js

```diff
@@ -1,12 +1,31 @@
 'use strict';
 
 var fs = require('fs');
+var path = require('path');
 
 var templatesTask = function (gulp, plugins, config, helpers) {
 
+  function getTemplateInputData(file) {
+    try {
+      var data = JSON.parse(
+        fs.readFileSync(config.src.dataPath + path.basename(file.path) + '.json')
+      );
+    } catch(e) {
```

Comment:
I'm just not an expert in Node I/O, but what you're saying seems logical. Simpile benchmarks on this would be informative (just out of curiosity, not needed for this PR). By the order I was referring to the situation, where we have twigs and jsons loaded into gulp stream. I think we could have a situation where gulp loads templ1.twig, then templ1.json, then process the templ1.json data to the next templ2.twig and so on.